capabilities/service.go

```diff
@@ -0,0 +1,101 @@
+// Package capabilities provides an service to find out which capabilities/functions
+// a tenant cluster on the given installation will provide.
+package capabilities
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/gsctl/client"
+)
+
+// Service provides methods to get more details on the installation's
+// and tenant cluster's capabilities.
+type Service struct {
+	// Provider identifies the installation's provide,r e. g. 'aws' or 'azure'.
+	Provider string
+
```

Comment:
Here one should keep internal copy of initialized capabilities because it's this specific `Service` instance that initializes them and also that is used to get them and provide functionality to check whether specific compatibility exists.